tools/trimmomatic.wdl

```
@@ -0,0 +1,151 @@
+        cpu: 8
+        maxRetries: 0
+    }
+}
+
+task trim_paired_end {
+    meta {
+        description: "Performs paired end trimming of reads using the Trimmomatic tool."
+        outputs: {
+            trimmed_reads_forward: "The forward trimmed paired-end read in FASTQ format."
+            trimmed_reads_reverse: "The reverse trimmed paired-end read in FASTQ format."
+        }
+    }
+
+    parameter_meta {
+        input_file_forward: "Forward input FASTQ format file to run Trimmomatic on."
+        input_file_reverse: "Reverse input FASTQ format file to run Trimmomatic on."
+        output_name: "Output FASTQ format file of trimmed reads."
+        leading: "Cut bases off the start of a read, if below a threshold quality."
+        trailing: "Cut bases off the end of a read, if below a threshold quality."
+        window_size: "Specifies the number of bases to average across for the sliding window."
+        window_quality: "Specifies the average quality required in the sliding window."
+        minlen: "Drop the read if it is below a specified length."
+    }
+
+    input {
+        File input_file_forward
+        File input_file_reverse
+        String outfile_name_forward = (
+            basename(input_file_forward, ".fastq.gz") + ".forward.trimmed.fastq.gz"
+        )
+        String outfile_name_reverse = (
+            basename(input_file_reverse, ".fastq.gz") + ".reverse.trimmed.fastq.gz"
+        )
+         String prefix = sub(
+            basename(input_file_forward),
+            "([_\\.][rR][12])?(\\.subsampled)?\\.(fastq|fq)(\\.gz)?$",
+            ""
+        )
+        String illumina_clip
+        Int? crop
+        Int? headcrop
+        Int? leading
+        Int minlen
+        Int threads = 8
+        Int? trailing
+        Int window_size
+        Int window_quality
+        Boolean phred64
+    }
+
+    command <<<
+        ls -l /usr/local/share/trimmomatic-0.36-5/adapters/
```

All our commands that contain more than one Bash statement should start with the line `set -euo pipefail`. This causes the script to fail in some common error cases instead of silently continuing and being hard to debug. (I recommend starting all shell scripts with that line.)

---

Actually, since it looks like you can remove this `ls` and the `mv`s at the bottom, this _will_ just be one Bash statement eventually. In which case `set -euo pipefail` won't do anything. You can leave it out, but still recommend setting that for anything more complicated.

---

Delete this `ls` now that you've shared the results of it